src/check_update.ts

```diff
@@ -82,43 +82,102 @@ async function createVersionMap() {
   return versionMap;
 }
 
-/** readProjectDependencies reads from possible dependency files
- * (import_map.json, slack.json) and maps them to the versionMap
- * containing each dependency's update information
+/** readProjectDependencies cycles through supported project
+ * dependency files and maps them to the versionMap that contains
+ * each dependency's update information.
  */
-async function readProjectDependencies(): Promise<VersionMap> {
+export async function readProjectDependencies(): Promise<VersionMap> {
   const cwd = Deno.cwd();
   const versionMap: VersionMap = {};
+  const dependencyFiles = await gatherDependencyFiles(cwd);
 
-  // Find SDK component versions in import map, if available
-  const map = await getJSON(`${cwd}/import_map.json`);
-  for (const sdkUrl of Object.values(map.imports) as string[]) {
-    for (const sdk of IMPORT_MAP_SDKS) {
-      if (sdkUrl.includes(sdk)) {
-        versionMap[sdk] = {
-          name: sdk,
-          current: extractVersion(sdkUrl),
-        };
+  for (const [file, depKey] of dependencyFiles) {
+    const fileJSON = await getJSON(`${cwd}/${file}`);
+    const fileDependencies = extractDependencies(fileJSON, depKey);
+
+    // For each dependency found, compare to SDK-related dependency
+    // list and, if known, update the versionMap with version information
+    for (const [_, val] of fileDependencies) {
+      for (const sdk of [...IMPORT_MAP_SDKS, ...SLACK_JSON_SDKS]) {
+        if (val.includes(sdk)) {
+          versionMap[sdk] = {
+            name: sdk,
+            current: extractVersion(val),
+          };
+        }
       }
     }
   }
 
-  // Find SDK component versions in slack.json, if available
-  const { hooks }: { [key: string]: string } = await getJSON(
-    `${cwd}/slack.json`,
-  );
-  for (const command of Object.values(hooks)) {
-    for (const sdk of SLACK_JSON_SDKS) {
-      if (command.includes(sdk)) {
-        versionMap[sdk] = {
-          name: sdk,
-          current: extractVersion(command),
-        };
-      }
+  return versionMap;
+}
+
+/**
+ * gatherDependencyFiles rounds up all SDK-supported dependency files, as well
+ * as those dependency files referenced in deno.json or deno.jsonc, and returns
+ * an array of arrays made up of filename and dependency key pairs.
+ */
+export async function gatherDependencyFiles(cwd: string): Promise<string[][]> {
+  const dependencyFiles = [
+    ["import_map.json", "imports"],
```

Comment:
I wonder if we could drop checking for this file explicitly. If the user doesn't have a `deno.(json|jsonc)` file that has an `importMap` entry, it isn't being referenced. Could we rely on the import map being defined in the deno config file and it would get added to the list then?

Reply:
Hm. Are you saying that it's not possible for a user to just have an `import_map(.json|.jsonc)` file going forward? I was under the impression that most folks' projects that we have today feature that setup, and that the optional reference in `deno.json` via `importMap` was an alternative.

If so, then yeah, we can remove the explicit listing of `import_map.json`, although one could potentially put any name under the `importMap` value, so it wouldn't necessarily be `import_map.json`.

Reply:
All of our hooks use `--config=deno.jsonc` and the importMap location is defined in that config file. For a developer to change that, they'll need to remove `importMap` from the config file and pass in the import map location via `--import_map=...` which I don't think is a use case we need to worry about personally

Reply:
Yah, what Neil said 😄 - an import map is only in effect if it's defined in the deno.json file, and they can name it w/e they want, so if we're detecting `importMap` in a deno.json(c) file, and looking for that, then we're good (which you're PR is already doing so that's awesome).

Reply:
Ah, of course! 🤦🏼‍♀️ Thanks for taking the time to explain this. Ok, I'll remove it.